src/main/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticator.java

```diff
@@ -109,8 +116,16 @@ public HTTPJwtAuthenticator(final Settings settings, final Path configPath) {
         jwtHeaderName = settings.get("jwt_header", HttpHeaders.AUTHORIZATION);
         isDefaultAuthHeader = HttpHeaders.AUTHORIZATION.equalsIgnoreCase(jwtHeaderName);
         rolesKey = settings.get("roles_key");
+        jsonRolesPath = settings.get("roles_path");
         subjectKey = settings.get("subject_key");
         jwtParser = _jwtParser;
+
+        // throw exception when settings provided both roles_key and roles_path
+        if (rolesKey != null && jsonRolesPath != null) {
+            throw new IllegalStateException("Both roles_key and roles_path are simultaneously provided." +
```

Comment:
Can you check what other implementations of `HTTPAuthenticator` throw when settings are not what they expect?